bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/DataClientClassicApi.java

```diff
@@ -0,0 +1,235 @@
+import com.google.cloud.bigtable.grpc.scanner.ResultScanner;
+import com.google.cloud.bigtable.hbase.adapters.Adapters;
+import com.google.cloud.bigtable.hbase.wrappers.BulkMutationWrapper;
+import com.google.cloud.bigtable.hbase.wrappers.BulkReadWrapper;
+import com.google.cloud.bigtable.hbase.wrappers.DataClientWrapper;
+import com.google.cloud.bigtable.util.ApiFutureUtil;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.grpc.stub.StreamObserver;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.hadoop.hbase.client.Result;
+
+public class DataClientClassicApi implements DataClientWrapper {
+
+  private final BigtableSession session;
+  private final BigtableDataClient delegate;
+  private final RequestContext requestContext;
+
+  DataClientClassicApi(BigtableSession session, RequestContext requestContext) {
+    this.session = session;
+    this.delegate = session.getDataClient();
+    this.requestContext = requestContext;
+  }
+
+  @Override
+  public BulkMutationWrapper createBulkMutation(String tableId) {
+    BigtableTableName tableName =
+        new BigtableTableName(
+            NameUtil.formatTableName(
+                requestContext.getProjectId(), requestContext.getInstanceId(), tableId));
+    return new BulkMutationClassicApi(session.createBulkMutation(tableName));
+  }
+
+  @Override
+  public BulkReadWrapper createBulkRead(String tableId) {
+    BigtableTableName tableName =
+        new BigtableTableName(
+            NameUtil.formatTableName(
+                requestContext.getProjectId(), requestContext.getInstanceId(), tableId));
+    return new BulkReadClassicApi(session.createBulkRead(tableName));
+  }
+
+  @Override
+  public ApiFuture<Void> mutateRowAsync(RowMutation rowMutation) {
+    ListenableFuture<MutateRowResponse> response =
+        delegate.mutateRowAsync(rowMutation.toProto(requestContext));
+    return ApiFutureUtil.transformAndAdapt(
+        response,
+        new Function<MutateRowResponse, Void>() {
+          @Nullable
+          @Override
+          public Void apply(@Nullable MutateRowResponse mutateRowResponse) {
+            return null;
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Result> readModifyWriteRowAsync(ReadModifyWriteRow readModifyWriteRow) {
+    ListenableFuture<ReadModifyWriteRowResponse> response =
+        delegate.readModifyWriteRowAsync(readModifyWriteRow.toProto(requestContext));
+    return ApiFutureUtil.transformAndAdapt(
+        response,
+        new Function<ReadModifyWriteRowResponse, Result>() {
+          @Override
+          public Result apply(ReadModifyWriteRowResponse response) {
+            // TODO(rahulkql): should we create RowAdapter for proto Row to hbase's Result?
```

Comment:
I think so, it would avoid an unnecessary intermediate transformation

Reply:
Thanks for the suggestion, I would that in a separate PR.